### PR TITLE
Add support for static varaibles

### PR DESCRIPTION
**Pull request by jarzec**
This PR is my attempt at https://github.com/hsutter/cppfront/issues/212 with tests.

### CODE REVIEW

File: source/parse.h [hunk 1]
```diff
@@ -519,6 +519,7 @@ struct type_id_node
 {
     source_position pos;
 
+    token const* static_qualifier = {};
```
Comment (jarzec) [inline source/parse.h:522]: [Explanation] I could have easily included `static` in `pc_qualifiers` (that could be renamed). I decided against that as `static` is of somewhat different nature.

Comment (hsutter) [reply source/parse.h:522]: Yes, that's the way I tend to handle these qualifier.

But I feel bad that I'm going to close this PR unmerged now, seeing all the work you've put into it -- as I mentioned on #212 I'm still thinking about what I want to do for `static` variables and I'd like to push them down the road, especially to get experience with classes first (and how the demands for static members unfolds there)... sorry, and thanks for understanding!

**Comment by hsutter** [comment]
Closing for now per comment above and #212 -- thanks for understanding! I appreciate the work you put into this though and that you wrote the code thoughtfully and included lots of test cases.

**Comment by jarzec** [comment]
I understand and agree with your remarks. The work is not lost - I got to understand better how the code works.
BTW. do you already have a time frame for simple classes? I think this is the last bit missing before cppfront can be rewritten in Cpp2 😉.

**Comment by hsutter** [comment]
Now that I've caught up on issues and PRs (barring a new wave) I hope to be able to start looking at the next set of features... thanks again.

**Pull request closed without merging**